src/Sentry.Unity.Android/JniExecutor.cs

```diff
@@ -1,182 +1,58 @@
 using System;
-using System.Threading;
-using System.Threading.Tasks;
 using Sentry.Extensibility;
-using UnityEngine;
 
 namespace Sentry.Unity.Android;
 
-internal class JniExecutor : IJniExecutor
+internal class JniExecutor
 {
-    // We're capping out at 16ms - 1 frame at 60 frames per second
-    private static readonly TimeSpan DefaultTimeout = TimeSpan.FromMilliseconds(16);
-
-    private readonly CancellationTokenSource _shutdownSource;
-    private readonly AutoResetEvent _taskEvent;
     private readonly IDiagnosticLogger? _logger;
+    private readonly IAndroidJNI _androidJNI;
 
-    private Delegate _currentTask = null!; // The current task will always be set together with the task event
-
-    private TaskCompletionSource<object?>? _taskCompletionSource;
-
-    private readonly object _lock = new object();
-
-    private bool _isDisposed;
-    private Thread? _workerThread;
-
-    public JniExecutor(IDiagnosticLogger? logger)
+    public JniExecutor(IDiagnosticLogger? logger, IAndroidJNI? androidJNI = null)
     {
         _logger = logger;
-        _taskEvent = new AutoResetEvent(false);
-        _shutdownSource = new CancellationTokenSource();
-
-        _workerThread = new Thread(DoWork) { IsBackground = true, Name = "SentryJniExecutorThread" };
-        _workerThread.Start();
+        _androidJNI ??= androidJNI ?? AndroidJNIAdapter.Instance;
     }
 
-    private void DoWork()
+    public TResult? Run<TResult>(Func<TResult?> jniOperation, bool? isMainThread = null)
```

Comment:
is the argument here for testing purposes? I imagine we can't trust the caller on whether it's UI thread or not?

Reply:
Updated by splitting public/interal for testing purposes.